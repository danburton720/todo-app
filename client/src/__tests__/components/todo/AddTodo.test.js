import React from 'react';
import {mount} from 'enzyme';
import AddTodo from '../../../app/components/todo/AddTodo';

describe('AddTodo', () => {
    let mountWrapper;
    let props;
    const wrapper = () => {
        if (!mountWrapper) {
            mountWrapper = mount(<AddTodo {...props}/>);
        }
        return mountWrapper;
    };

    beforeEach(() => {
        props = {
            onSubmit: jest.fn(),
            cancel: jest.fn()
        };
        mountWrapper = undefined;
    });

    describe('render', () => {
        it('should return a form with an input and two buttons', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });

    describe('functionality', () => {
        describe('input', () => {
            it('should set the input value to \'hello world\'', () => {
                const onChangeEvent = {
                    target: {
                        value: 'hello world'
                    }
                };
                wrapper().find('input').props().onChange(onChangeEvent);
                wrapper().update();
                expect(wrapper().find('input').props().value).toBe('hello world')
            });
        });
        describe('form', () => {
            describe('onSubmit', () => {
                it('should call the prop onSubmit and clear the input', () => {
                    const onChangeEvent = {
                        target: {
                            value: 'hello world'
                        }
                    };
                    const submitEvent = {
                        preventDefault: jest.fn()
                    };
                    wrapper().find('input').props().onChange(onChangeEvent);
                    wrapper().update();
                    expect(wrapper().find('input').props().value).toBe('hello world');
                    wrapper().find('form').props().onSubmit(submitEvent);
                    expect(submitEvent.preventDefault).toHaveBeenCalledTimes(1);
                    wrapper().update();
                    expect(props.onSubmit).toHaveBeenCalledTimes(1);
                    expect(props.onSubmit).toHaveBeenCalledWith({
                        description: 'hello world'
                    });
                    expect(wrapper().find('input').props().value).toBe('');
                });
            });
        });
    });
});