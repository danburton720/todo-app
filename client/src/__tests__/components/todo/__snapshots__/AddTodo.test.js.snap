// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo render should return a form with an input and two buttons 1`] = `
<AddTodo
  cancel={[MockFunction]}
  onSubmit={[MockFunction]}
>
  <form
    className="addTodoForm"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="add new todo list"
      value=""
    />
    <div
      className="addTodoButtonWrapper"
    >
      <button
        className="primary"
        type="submit"
      >
        Create
      </button>
      <button
        className="primary inverted"
        onClick={[MockFunction]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</AddTodo>
`;
